File: sqlite-android-studio-plugin/src/main/java/com.alecstrong.sqlite.android/lang/SqliteFileTypeFactory.java
package com.alecstrong.sqlite.android.lang;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;

public class SqliteFileTypeFactory extends FileTypeFactory {
	@Override
	public void createFileTypes(FileTypeConsumer consumer) {
		consumer.consume(SqliteFileType.INSTANCE, SqliteFileType.DEFAULT_EXTENSION);
	}
}

File: sqlite-android-studio-plugin/src/main/kotlin/com/alecstrong/sqlite/android/lang/SqliteFileTypeFactory.kt
package com.alecstrong.sqlite.android.lang

import com.alecstrong.sqlite.android.SqliteCompiler
import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

public class SqliteFileTypeFactory : FileTypeFactory() {
  override fun createFileTypes(consumer: FileTypeConsumer) {
    consumer.consume(SqliteFileType.INSTANCE, SqliteCompiler.FILE_EXTENSION)
  }
}
